content/blog/text-rendering-updates/index.Rmd

```diff
@@ -0,0 +1,171 @@
+* [x] Find photo & update yaml metadata
+* [x] Create `thumbnail-sq.jpg`; height and width should be equal
+* [x] Create `thumbnail-wd.jpg`; width should be >5x height
+* [x] `hugodown::use_tidy_thumbnails()`
+* [ ] Add intro sentence, e.g. the standard tagline for the package
+* [ ] `usethis::use_tidy_thanks()`
+-->
+
+```{r, setup, include=FALSE}
+knitr::opts_chunk$set(
+  dev = "ragg_png"
+)
+```
+
+<blockquote class="bluesky-embed" data-bluesky-uri="at://did:plc:6cf4jj62ofgoswlxcqdvtcg5/app.bsky.feed.post/3lfevs4qe5k2l" data-bluesky-cid="bafyreigvpe72rfvuw47nr6qrec2je2lntsowdnzer5vz2imjbejree6ebm"><p lang="en">text rendering is one of those disciplines where, if you think you finally got it right, you can be 100% certain that you didn&#x27;t</p>&mdash; Thomas Lin Pedersen (<a href="https://bsky.app/profile/did:plc:6cf4jj62ofgoswlxcqdvtcg5?ref_src=embed">@thomasp85.com</a>) <a href="https://bsky.app/profile/did:plc:6cf4jj62ofgoswlxcqdvtcg5/post/3lfevs4qe5k2l?ref_src=embed">January 10, 2025 at 10:44 AM</a></blockquote><script async src="https://embed.bsky.app/static/embed.js" charset="utf-8"></script>
+
+No reason to hide the fact: Text rendering is complicated! When I set out to improve the support for modern text rendering features in R all those years ago, I don't think I truly appreciated that fact. And probably for the better, since I'm not sure I would have taken on the task.
+
+Taking the quote above as a universal truth (it comes from a reputable source after all), I'm sure I'll never be fully done, but recent work on the whole stack at least makes me worry less about the correctness. This post will go through the changes that span the [systemfonts](https://github.com/r-lib/systemfonts), [textshaping](https://github.com/r-lib/textshaping), and [marquee](https://marquee.r-lib.org) packages and let you now how you, as a user or developer, should take advantage of them.
+
+## Working with non-installed fonts
+
+The genesis of the systemfonts package was a need to be able to tap into the operating systems font library so that whatever was installed on the system would be available to graphics devices (assuming those devices used systemfonts). The scope of the package has gradually increased and one of the needs that has become obvious over time is a way to work with fonts that aren't installed on the system (E.g. if you want to bundle a font with a package, or if you are deploying a Shiny app that uses a specific font for the graphics). Until now the `register_font()` and `register_variant()` functions have been the only option for letting systemfonts know about fonts other than those installed on the system. However, both of these functions were designed to circumvent limitations in the R graphics system when it comes to font selection (e.g. no way to use a "thin" font variant as the only weight option in the graphics system is bold yes/no) and as such were clunky to use for introducing new fonts.
```

Comment:
Before diving into the technical details, let's understand why font management matters: when creating visualizations or applications, you often need specific fonts to maintain consistent branding or improve readability. And until recently, this was surprisingly difficult in R. The systemfonts package originally focused on accessing your operating system's font library. However, modern development often requires working with fonts that aren't installed system-wide: think of bundling fonts with a package or deploying a Shiny app with custom typography.